server/src/instant/storage/routes.clj

```diff
@@ -23,63 +22,57 @@
         objects (:object-summaries objects-resp)]
     (response/ok {:data (map #(select-keys % [:key :size :owner :etag]) objects)})))
 
-
 (comment
   (def b s3-util/default-bucket)
   (buckets-get {})
   (objects-get {:params {:bucket b}}))
 
-(comment
-  (def rules {:code
-              {"$files" {"bind" ["isLoggedIn" "auth.id != null"]
-                         "allow" {"view" "isLoggedIn"
-                                  "create" "data.path.contains('demo')"}}}})
-  (def program (rule-model/get-program! rules "$files" "create"))
-  (cel/eval-program! program {"auth" (cel/->cel-map {"id" "alex"})
-                              "data" (cel/->cel-map {"path" "demo/image.png"})}))
-
-(defn assert-storage-permission! [action {:keys [app-id refresh-token filepath]}]
-  (let [rules (rule-model/get-by-app-id {:app-id app-id})
-        current-user (when refresh-token
-                       (app-user-model/get-by-refresh-token {:app-id app-id
-                                                             :refresh-token refresh-token}))
-        program (rule-model/get-program! rules "$files" action)]
-    (ex/assert-permitted!
-     :has-storage-permission?
-     ["$files" action]
-     (if-not program
-       ;; deny access by default if no permissions are currently set
-       false
-       ;; otherwise, evaluate the permissions code
-       (cel/eval-program! program
-                          {"auth" (cel/->cel-map (<-json (->json current-user)))
-                           "data" (cel/->cel-map {"path" filepath})})))))
+(defn req->app-file! [req params]
+  (let [id (ex/get-param! params [:app_id] uuid-util/coerce)
+        {app-id :id} (app-model/get-by-id! {:id id})
+        filename (ex/get-param! params [:filename] string-util/coerce-non-blank-str)
```

Comment:
Might validate the `filename` param before fetching the app, so that you don't have to do the extra work of fetching the app if the filename param is invalid.